<?xml version="1.0" encoding="utf-8" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/etudiant/rosIn_ws/src/motoman/motoman_hc10_support/urdf/hc10.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="motoman_hc10">

    <!-- Properties -->
    
    <material name="yaskawa_blue">
        <color rgba="0.12941 0.14902 0.74902 1" />
    </material>

    <!-- link list -->

    <link name="base_link">
        <visual>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/visual/base_link.dae" />
            </geometry>
            <material name="yaskawa_blue" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/collision/base_link.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>

    <link name="link_1_s">
        <visual>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/visual/link_1_s.dae" />
            </geometry>
            <material name="yaskawa_blue" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/collision/link_1_s.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>

    <link name="link_2_l">
        <visual>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/visual/link_2_l.dae" />
            </geometry>
            <material name="yaskawa_blue" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/collision/link_2_l.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>

    <link name="link_3_u">
        <visual>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/visual/link_3_u.dae" />
            </geometry>
            <material name="yaskawa_blue" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/collision/link_3_u.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>

    <link name="link_4_r">
        <visual>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/visual/link_4_r.dae" />
            </geometry>
            <material name="yaskawa_blue" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/collision/link_4_r.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>

    <link name="link_5_b">
        <visual>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/visual/link_5_b.dae" />
            </geometry>
            <material name="yaskawa_blue" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/collision/link_5_b.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>

    <link name="link_6_t">
        <visual>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/visual/link_6_t.dae" />
            </geometry>
            <material name="yaskawa_blue" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://motoman_hc10_support/meshes/hc10/collision/link_6_t.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>

    <link name="tool0" />

    <!-- end of link list -->

    <!-- joint list -->
    
    <joint name="fixed" type="fixed">
        <parent link="world" />
        <child link="base_link" />
    </joint>
    
    <joint name="joint_1_s" type="revolute">
        <parent link="base_link" />
        <child link="link_1_s" />
        <origin rpy="0 0 0" xyz="0 0 0.275" />
        <axis xyz="0 0 1" />
        <limit effort="368.48" lower="-3.14159265359" upper="3.14159265359" velocity="2.26892802759" />
    </joint>
    <joint name="joint_2_l" type="revolute">
        <parent link="link_1_s" />
        <child link="link_2_l" />
        <origin rpy="0 0 0" xyz="0 0 0" />
        <axis xyz="0 1 0" />
        <limit effort="368.48" lower="-3.14159265359" upper="3.14159265359" velocity="2.26892802759" />
    </joint>
    <joint name="joint_3_u" type="revolute">
        <parent link="link_2_l" />
        <child link="link_3_u" />
        <origin rpy="0 0 0" xyz="0 0 0.700" />
        <axis xyz="0 -1 0" />
        <limit effort="141.12" lower="-0.0872664625997" upper="6.19591884458" velocity="3.14159265359" />
    </joint>
    <joint name="joint_4_r" type="revolute">
        <parent link="link_3_u" />
        <child link="link_4_r" />
        <origin rpy="0 0 0" xyz="0 0 -0.500" />
        <axis xyz="0 0 1" />
        <limit effort="39.20" lower="-3.14159265359" upper="3.14159265359" velocity="3.14159265359" />
    </joint>
    <joint name="joint_5_b" type="revolute">
        <parent link="link_4_r" />
        <child link="link_5_b" />
        <origin rpy="0 0 0" xyz="0 0.162 0" />
        <axis xyz="0 -1 0" />
        <limit effort="31.36" lower="-3.14159265359" upper="3.14159265359" velocity="4.36332312999" />
    </joint>
    <joint name="joint_6_t" type="revolute">
        <parent link="link_5_b" />
        <child link="link_6_t" />
        <origin rpy="0 0 0" xyz="0 0 0" />
        <axis xyz="0 0 1" />
        <limit effort="31.36" lower="-3.14159265359" upper="3.14159265359" velocity="4.36332312999" />
    </joint>
    <joint name="joint_6_t-tool0" type="fixed">
        <origin rpy="3.14159265359 0 0" xyz="0 0 -0.130" />
        <parent link="link_6_t" />
        <child link="tool0" />
    </joint>

    <!-- end of joint list -->
    
    <transmission name="trans_joint_1_s">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_1_s">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_1_s_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    <transmission name="trans_joint_2_l">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_2_l">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_2_l_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    <transmission name="trans_joint_3_u">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_3_u">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_3_u_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    <transmission name="trans_joint_4_r">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_4_r">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_4_r_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    <transmission name="trans_joint_5_b">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_5_b">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_5_b_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    <transmission name="trans_joint_6_t">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_6_t">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_6_t_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    <gazebo>
        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
            <robotNamespace>/motoman_hc10</robotNamespace>
            <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
        </plugin>
    </gazebo>
</robot>
